internal/webhooks/clusterclass_test.go

```diff
@@ -1318,6 +1320,854 @@ func TestClusterClassValidationWithClusterAwareChecks(t *testing.T) {
+								Nullable: true,
+							},
+						},
+					},
+					clusterv1.ClusterClassVariable{
+						Name: "hdd",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type:     "number",
+								Nullable: true,
+							},
+						},
+					},
+				).
+				Build(),
+			newClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "cdrom",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type:     "number",
+								Nullable: true,
+							},
+						},
+					},
+					clusterv1.ClusterClassVariable{
+						Name: "hdd",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type:     "number",
+								Nullable: true,
+							},
+						},
+					}).
+				Build(),
+			expectErr: true,
+		},
+
+		{
+			name: "Error if required variable is added but not defined in all clusters",
+			clusters: []client.Object{
+				builder.Cluster(metav1.NamespaceDefault, "cluster1").
+					WithLabels(map[string]string{clusterv1.ClusterTopologyOwnedLabel: ""}).
+					WithTopology(
+						builder.ClusterTopology().
+							WithClass("class1").
+							WithVariables(
+								clusterv1.ClusterVariable{
+									Name: "cpu",
+									Value: apiextensionsv1.JSON{
+										Raw: []byte(`4`),
+									},
+								}).
+							Build()).
+					Build(),
+				builder.Cluster(metav1.NamespaceDefault, "cluster2").
+					WithLabels(map[string]string{clusterv1.ClusterTopologyOwnedLabel: ""}).
+					WithTopology(
+						builder.ClusterTopology().
+							WithClass("class1").
+							WithVariables(
+								clusterv1.ClusterVariable{
+									Name: "cpu",
+									Value: apiextensionsv1.JSON{
+										Raw: []byte(`4`),
+									},
+								},
+								clusterv1.ClusterVariable{
+									Name: "hdd",
+									Value: apiextensionsv1.JSON{
+										Raw: []byte(`"hdd0"`),
+									},
+								},
+							).
+							Build()).
+					Build(),
+			},
+			oldClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type: "integer",
+							},
+						},
+					},
+				).
+				Build(),
+			newClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name:     "hdd",
+						Required: true,
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type: "string",
+							},
+						},
+					},
+					clusterv1.ClusterClassVariable{
+						Name: "cpu",
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type: "integer",
+							},
+						},
+					}).
+				Build(),
+			expectErr: true,
+		},
+		{
+			name: "Error if update to make variable required is rejected if not defined in all clusters",
```

Comment:
This reads slightly confusing. Is it essentially the same as the case above with more clusters/variables?

Reply:
Yeah - I'll add a comment, but I think it's good to have a test case that does a little bit of stretching of our array etc. logic.

Reply:
Yup that's fine. It's just that this name is constructed relatively complicated compared to the one above. (error and rejected sounds redundant and makes it (relatively) hard to parse)